debezium-core/src/main/java/io/debezium/relational/mapping/MaskStrings.java

```diff
@@ -142,12 +138,8 @@ public Object convert(Object value) {
         private String toHash(Serializable value) throws IOException {
             hashAlgorithm.reset();
             hashAlgorithm.update(salt);
-
-            try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
-                    ObjectOutput out = new ObjectOutputStream(bos)) {
-                out.writeObject(value);
-                return convertToHexadecimalFormat(hashAlgorithm.digest(bos.toByteArray()));
-            }
+            byte[] data = value.toString().getBytes();
```

Comment:
So I got to admit it's completely unclear to me why we'd ever use that approach via serialization through `BAOS` here instead of doing the simple conversion you propose. There's a problem though in terms of compatibility: we cannot simply change the implementation, as that may break existing users who'd suddenly see different hash values for the same original values (as indicated by your changes to the assertions in the test), defeating the purpose of this _consistent_ hashing feature.

So I'm afraid we'll have to add a connector configuration allowing the user to opt into this new behavior. The old behavior would remain the default, and we'd change that (and drop the option and the old implementation in a potential Debezium 2.0 version at some point in the future).

Reply:
In fact, the data hashed with the previous code is not validated in any way. Because somehow the non-characters got into the work and were transferred. According to the version, a code thread that exhibits different behavior can be added here. Whatever you say that get in the way.  :) @gunnarmorling

Reply:
That's the part I don't quite get yet; sure the serialization step adds some special bytes (and I agree it's weird to do it the way it's currently done), but they don't show up as such in the final hex-encoded value, do they? If you look at the assertion you modified, it's a regular hex value as any other. So I'm curious why this actually is a problem in your case? Is it that you expect the particular hash value, so you can e.g. compare/correlate it to the same value being hashed the same way elsewhere?

Reply:
I'm giving an example of how to get user information that we hash here. We collect data from many sources with Debezium. We want to check the singularity of the data. However, although we use the same salt, the same hash does not occur. The mappings are not working correctly. 

The byte array consisting of the current code: [-84, -19, 0, 5, 116, 0, 22, 117, 110, 97, 108, 46, 115, 117, 114, 109, 101, 108, 105, 64, 100, .............]

The byte array that must be : [117, 110, 97, 108, 46, 115, 117, 114, 109, 101, 108, 105, 64, 100, .............]

Reply:
@gunnarmorling I've added a version. Can you review it?

Reply:
Ok, I see; so you need fidelity for the value being hashed the same way in different places. Makes sense of course. Thanks for clarifying!